servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -891,4 +918,166 @@ public static StacklessConnectionRejectedException newInstance(String message, C
             return ThrowableUtils.unknownStackTrace(new StacklessConnectionRejectedException(message), clazz, method);
         }
     }
+
+    private static boolean isClosedList(List<?> list) {
+        return list.getClass().equals(ClosedList.class);
+    }
+
+    private static final class ClosedList<T> implements List<T> {
+        private final List<T> delegate;
+
+        private ClosedList(final List<T> delegate) {
+            this.delegate = requireNonNull(delegate);
+        }
+
+        @Override
+        public int size() {
+            return delegate.size();
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return delegate.isEmpty();
+        }
+
+        @Override
+        public boolean contains(final Object o) {
+            return delegate.contains(o);
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            return delegate.iterator();
+        }
+
+        @Override
+        public void forEach(final Consumer<? super T> action) {
+            delegate.forEach(action);
+        }
+
+        @Override
+        public Object[] toArray() {
+            return delegate.toArray();
+        }
+
+        @Override
+        public <T1> T1[] toArray(final T1[] a) {
```

Comment:
There is also `toArray(final IntFunction<T1[]> generator)` overload that is not delegated. Also, consider delegating methods from `Object`

Reply:
`toArray(final IntFunction<T1[]> generator)` -> this is JDK11 only so we can't add it.